corehq/apps/reports/views.py

```diff
@@ -527,9 +527,11 @@ def email_report(request, domain, report_slug, dispatcher_class=ProjectReportDis
         recipient_emails.add(request.couch_user.get_email())
 
     request_data = request_as_dict(request)
-
     report_type = dispatcher_class.prefix
-    send_email_report.delay(recipient_emails, domain, report_slug, report_type,
+    datespan = request_data.pop('datespan')
+    request_data['startdate'] = datespan.startdate.isoformat()
+    request_data['enddate'] = datespan.enddate.isoformat()
+    send_email_report.delay(list(recipient_emails), domain, report_slug, report_type,
```

Comment:
nope, sets are not json serializable from what I've seen. I'll add a small comment here

Reply:
I see that [`send_email_report`'s docstring](https://github.com/dimagi/commcare-hq/blob/7b5efafd81d7f3d97a60898586c3192c5ef9ff8e/corehq/apps/saved_reports/tasks.py#L113) has `recipient_emails` name as `recipient_list` -- which is a better name, because it hints that the function is expecting a list.

(We should remove that list of parameters from the docstring. All it's doing is explaining their data types, and the only types that are not obvious are `recipient_emails` and `once`. Renaming it `is_once_off` would make the type obvious there too.)

(The reason `recipient_emails` is a set here is to remove duplicates. That wouldn't be necessary if `send_email_report` did its own due diligence. Personally, I would go with
```python
    recipient_list = form.cleaned_data['recipient_emails']
    if form.cleaned_data['send_to_owner']:
        recipient_list.append(request.couch_user.get_email())
```
and then remove duplicates in `send_email_report`. It would be a little simpler, a little more obvious, and there would be no need for comments. (I appreciate this is way out of scope. If I have some time I might just do this a bit later myself.))